src/components/PageReady/index.js

```diff
@@ -1,15 +1,12 @@
 import React from 'react';
-import axios from 'axios';
 import { interpretErrorMessage } from './formatErrors';
 import type RequestShape from './request';
 
 import defaultRequest from './request';
 
 type PropShape = {
   predefinedAction?: Object,
-  url?: string,
-  postBody?: Object,
-  method?: string,
+  requestInstance: Promise,
```

Comment:
my first thought is to instead: pass in a function that returns a promise

Reply:
done